server/src/main/java/org/elasticsearch/common/geo/LinearRingEdgeTreeReader.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.geo;
+
+import org.apache.lucene.index.PointValues;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.io.stream.ByteBufferStreamInput;
+import org.elasticsearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import static org.apache.lucene.geo.GeoUtils.lineRelateLine;
+
+public class LinearRingEdgeTreeReader {
+    final BytesRef bytesRef;
+
+    public LinearRingEdgeTreeReader(BytesRef bytesRef) {
+        this.bytesRef = bytesRef;
+    }
+
+    /**
+     * even partially
+     */
+    public boolean containedIn(int minX, int minY, int maxX, int maxY) throws IOException {
+        StreamInput input = new ByteBufferStreamInput(ByteBuffer.wrap(bytesRef.bytes, bytesRef.offset, bytesRef.length));
```

Comment:
I am confused with the name of this method, shouldn't be intersects? maybe add in the java docs what this method should do.

Reply:
The confusion is justified because I was very confused myself. I initially thought that
I can do contains and crosses in the same pass of the tree. I was wrong.

I will split out the two methods and add more documentation!